lib/core/src/Cardano/DB/Sqlite.hs

```diff
@@ -195,14 +199,23 @@ startSqliteBackend migrateAll trace fp = do
             pure $ Left e
         Right _ -> pure $ Right ctx
 
--- | Exception predicate for migration errors.
-isMigrationError :: PersistException -> Maybe MigrationError
-isMigrationError e
-    | mark `isInfixOf` msg = Just $ MigrationError $ T.pack msg
-    | otherwise = Nothing
-  where
-    msg = show e
-    mark = "Database migration: manual intervention required."
+class Exception e => IsMigrationError e where
+    -- | Exception predicate for migration errors.
+    isMigrationError :: e -> Maybe MigrationError
```

Comment:
I think the function name `isMigrationError` is a bit misleading, as the prefix `is` strongly implies a Boolean result, when what we actually want is to _match and extract_ an error of a particular type (if it exists).

**Suggestion**: rename this to something like `catchMigrationError` or `matchMigrationError`.